src/core/hle/service/apt/applet_manager.cpp

```diff
@@ -1201,46 +1201,56 @@ void AppletManager::EnsureHomeMenuLoaded() {
 
 static void CaptureFrameBuffer(Core::System& system, u32 capture_offset, VAddr src, u32 height,
                                u32 format) {
-    static constexpr auto screen_capture_base_vaddr = static_cast<VAddr>(0x1F500000);
-    static constexpr auto screen_width = 240;
-    static constexpr auto screen_width_pow2 = 256;
+    if (format == UINT32_MAX) {
+        // This format is used to indicate the framebuffer should not be copied.
+        return;
+    }
+
+    // Address in VRAM that APT copies framebuffer captures to.
+    static constexpr auto screen_capture_base_vaddr = Memory::VRAM_VADDR + 0x500000;
     const auto bpp = format < 2 ? 3 : 2;
 
-    Memory::RasterizerFlushVirtualRegion(src, screen_width * height * bpp,
+    Memory::RasterizerFlushVirtualRegion(src, GSP::FRAMEBUFFER_WIDTH * height * bpp,
                                          Memory::FlushMode::Flush);
 
     auto dst_vaddr = screen_capture_base_vaddr + capture_offset;
     auto dst_ptr = system.Memory().GetPointer(dst_vaddr);
+    if (!dst_ptr) {
+        LOG_ERROR(Service_APT,
+                  "Could not retrieve framebuffer capture destination buffer, skipping screen.");
+        return;
+    }
+
     const auto src_ptr = system.Memory().GetPointer(src);
+    if (!src_ptr) {
+        LOG_ERROR(Service_APT,
+                  "Could not retrieve framebuffer capture source buffer, skipping screen.");
+        return;
+    }
+
     for (u32 y = 0; y < height; y++) {
-        for (u32 x = 0; x < screen_width; x++) {
-            auto dst_offset =
-                VideoCore::GetMortonOffset(x, y, bpp) + (y & ~7) * screen_width_pow2 * bpp;
-            auto src_offset = bpp * (screen_width * y + x);
+        for (u32 x = 0; x < GSP::FRAMEBUFFER_WIDTH; x++) {
+            auto dst_offset = VideoCore::GetMortonOffset(x, y, bpp) +
```

Comment:
```suggestion
            const auto dst_offset = VideoCore::GetMortonOffset(x, y, bpp) +
```
ditto for some other small const correctness stuff